src/azure-cli/azure/cli/command_modules/role/_params.py

```diff
@@ -205,6 +205,35 @@ def load_arguments(self, _):
         c.argument('create_cert', action='store_true', arg_group='keyCredentials', help='Create a self-signed certificate to use for the credential')
         c.argument('keyvault', arg_group='keyCredentials', help='Name or ID of a KeyVault to use for creating or retrieving certificates.')
 
+    for item in ['app', 'sp']:
+        with self.argument_context(f'ad {item} federated-credential') as c:
+            # TODO: We have to decide which name to use
+            #   --credential-id is consistent with API
+            #   --key-id is consistent with passwordCredentials and keyCredentials
+            c.argument('federated_identity_credential_id_or_name', options_list=['--credential-id', '--key-id'],
+                       help='Name or ID of the federated identity credential')
+            c.argument('audiences', nargs='*',
+                       help='Space-separated audiences that can appear in the external token. This field is mandatory, and '
+                            'defaults to "api://AzureADTokenExchange". It says what Microsoft identity platform should '
+                            'accept in the aud claim in the incoming token. This value represents Azure AD in your '
+                            'external identity provider and has no fixed value across identity providers - you may need '
+                            'to create a new application registration in your identity provider to serve as the audience '
+                            'of this token.')
```

Comment:
We should decide whether `create` command should take JSON or flattened params:

- JSON: Consistent with manifest and REST API, but is difficult to input in terminal (may also trigger PowerShell issue  (https://github.com/Azure/azure-cli/issues/15529)
- Flattened params: Easy to type in command line, but user has to run create command multiple times to create multiple credentials

Reply:
why is it difficult?

Reply:
Typing a JSON by hand seems difficult to me. 😂